salt/orch/update-etc-hosts.sls

```diff
@@ -1,13 +1,18 @@
+{% if salt['mine.get']('roles:kube-(master|minion)', 'caasp_fqdn', expr_form='grain_pcre').values()|length > 0 %}
 update_pillar:
   salt.function:
-    - tgt: '*'
+    - tgt: 'roles:kube-(master|minion)'
+    - tgt_type: grain_pcre
     - name: saltutil.refresh_pillar
 
 update_grains:
   salt.function:
-    - tgt: '*'
+    - tgt: 'roles:kube-(master|minion)'
```

Comment:
Updated the PR with your comments, but I think we have the problem stated in the previous comment...